src/lib/api/feed-manip.ts

```diff
@@ -89,23 +92,33 @@ export class FeedViewPostsSlice {
       this.isOrphan = true
       return
     }
+    const root = reply.root
+    const rootIsView =
+      AppBskyFeedDefs.isPostView(root) || AppBskyFeedDefs.isBlockedPost(root)
+    // if parent is also the root, we have data!
+    const grandparent =
+      rootIsView && parent?.record?.reply?.parent?.uri === root?.uri
```

Comment:
Nit: This is overly defensive. Let's avoid using `?.` when we don't need it. For example, `root?.uri` is unnecessary because `root` always exists by now. This makes it obvious that we won't hit `undefined === undefined` by mistake — that's not obvious in the code as is.

`parent?.record?.reply` can be `parent.record.reply`.